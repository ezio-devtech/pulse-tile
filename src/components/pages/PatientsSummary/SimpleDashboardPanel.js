import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import { get } from 'lodash';

import { themeConfigs } from '../../../themes.config';

const SimpleDashboardPanel = ({ id, title, items, goToState, state, isHasPreview, isHasList, srcPrevirew, isFeeds }) => {
  const imageLink = (isFeeds && items.length > 0) ? items[0].link : state;
  let filterItemsArray = (items.length > 4) ? [{text: 'Loading ...'}, '', '', ''] : items;
  const isShowPagesSynopsisImages = get(themeConfigs, 'isShowPagesSynopsisImages', false);
  return (<div id={id} className="dashboard-item">
    <div className="board">
      <div className="board-header">
        <div className="control-group right">
          <button className="btn btn-success btn-inverse btn-board-more" aria-label="More" onClick={() => goToState(state)}><i className="btn-icon fa fa-caret-right" /></button>
        </div>
        <h2 className="board-title">{ title }</h2>
      </div>
      <div className="board-body">
        {(isHasPreview && isShowPagesSynopsisImages)
          ? <div
            className="board-preview"
            style={{ backgroundImage: `url(${srcPrevirew})` }}
            onClick={() => goToState(imageLink)}
          />
          : null
        }
        {isHasList
          ? <ul className="board-list">
            {filterItemsArray.map(item =>
              <li className="board-list-item" key={_.uniqueId('__SimpleDashboardPanel__item__')}>
                {item.text ? <a className="board-list-link"
                  onClick={() => goToState(`${state}/${item.sourceId}`, item.link)}
                  onKeyPress={(event) => {if( event.key == 'Enter' ){ goToState(`${state}/${item.sourceId}`, item.link) }} }
                  title={item.text}
                  tabIndex="0">
                    {item.text}
                  </a> : null}
              </li>)}
          </ul>
          : null}
      </div>
    </div>
  </div>)
};

SimpleDashboardPanel.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  isFeeds: PropTypes.bool,
  id: PropTypes.string,
};
SimpleDashboardPanel.defaultProps = {
  isFeeds: false,
  id: '',
};

export default SimpleDashboardPanel
