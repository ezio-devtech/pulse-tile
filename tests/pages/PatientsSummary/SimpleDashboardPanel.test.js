import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

import SimpleDashboardPanel from '../../../src/components/pages/PatientsSummary/SimpleDashboardPanel';

const testProps = {
  goToState: () => {},
  navigateTo: () => {},
  title: 'Contacts',
  items: [{
    source: 'Marand',
    sourceId: '07d7fbb8-88e7-4f69-b901-38395dd36a6b',
    text:	'John Smithy',
  }, {
    source: 'Marand',
    sourceId: '96e5537f-600e-4334-aaad-629ecfaeadf8',
    text: 'Andrew Finch',
  }],
  state: 'contacts',
};

describe('Component <SimpleDashboardPanel />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <SimpleDashboardPanel
        goToState={testProps.goToState}
        navigateTo={testProps.navigateTo}
        title={testProps.title}
        items={testProps.items}
        state={testProps.state}
      />);

    expect(component).toMatchSnapshot();

    expect(component.instance().props.goToState).toEqual(testProps.goToState);
    expect(component.instance().props.navigateTo).toEqual(testProps.navigateTo);
    expect(component.instance().props.title).toEqual(testProps.title);
    expect(component.instance().props.items).toEqual(testProps.items);
    expect(component.instance().props.state).toEqual(testProps.state);

    expect(component.find('.dashboard-item')).toHaveLength(1);
    expect(component.find('.board')).toHaveLength(1);
    expect(component.find('.board-title')).toHaveLength(1);
    expect(component.find('.board-title').text()).toEqual(testProps.title);
    expect(component.find('ul')).toHaveLength(1);
    expect(component.find('li')).toHaveLength(2);
    expect(component.find('button')).toHaveLength(1);
    component.find('button').simulate('click');
    component.find('li').at(0).simulate('click');
  });
});

