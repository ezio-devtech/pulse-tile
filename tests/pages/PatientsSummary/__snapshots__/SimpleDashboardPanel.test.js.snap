// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <SimpleDashboardPanel /> should renders with all props correctly 1`] = `
<div
  className="dashboard-item"
>
  <div
    className="board"
  >
    <div
      className="board-header"
    >
      <div
        className="control-group right"
      >
        <button
          className="btn btn-success btn-inverse btn-board-more"
          onClick={[Function]}
        >
          <i
            className="btn-icon fa fa-caret-right"
          />
        </button>
      </div>
      <h3
        className="board-title"
      >
        Contacts
      </h3>
    </div>
    <div
      className="board-body"
    >
      <ul
        className="board-list"
      >
        <li
          className="board-list-item"
          key="__SimpleDashboardPanel__item__1"
          onClick={[Function]}
        >
          <span
            className="board-list-link"
          >
            John Smithy
          </span>
        </li>
        <li
          className="board-list-item"
          key="__SimpleDashboardPanel__item__2"
          onClick={[Function]}
        >
          <span
            className="board-list-link"
          >
            Andrew Finch
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
